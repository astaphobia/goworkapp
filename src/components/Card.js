import React from 'react'
import PropTypes from 'prop-types'

import Background from '@/../images/bg-red.jpg'

class Card extends React.Component {
    static propTypes = {
        showBackground: PropTypes.bool,
        renderBody: PropTypes.func.isRequired,
        renderFooter: PropTypes.func
    }

    render() {
        const { showBackground, item, renderBody, renderFooter } = this.props
        return (
            <div className="card job-detail" style={{ margin: '0 0 20px 0' }}>
                {!!showBackground
                    && <img className="card-img-top" src={Background} height="160px" alt="background detail"/>
            || null}
                <div className="card-body">
                    {renderBody(item)}
                </div>
                {!!renderFooter
                    && (
                        <div className="card-footer text-muted">
                            {renderFooter(item)}
                        </div>
                    ) || null}
            </div>
        )
    }
}
export default Card