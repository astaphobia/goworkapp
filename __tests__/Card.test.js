import React from 'react'
import { shallow } from 'enzyme'

import Card from '@/components/Card'

describe('Card', () => {
    it('hope doing well Card', () => {
        const comp = shallow(<Card renderBody={() => {
            return <h1>Mock</h1>
        }}/>)
        expect(comp).toMatchSnapshot()
    })
})