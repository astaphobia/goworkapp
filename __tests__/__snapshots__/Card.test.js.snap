// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card hope doing well Card 1`] = `
<div
  className="card job-detail"
  style={
    Object {
      "margin": "0 0 20px 0",
    }
  }
>
  <div
    className="card-body"
  >
    <h1>
      Mock
    </h1>
  </div>
</div>
`;
